src/composite_algs.jl

```diff
@@ -1,4 +1,4 @@
-mutable struct AutoSwitch{nAlg,sAlg,tolType,T}
+mutable struct AutoSwitchCache{nAlg,sAlg,tolType,T}
```

Comment:
As mentioned in https://github.com/SciML/OrdinaryDiffEq.jl/issues/325, it would make sense to make this immutable.

Reply:
Never mind. I see that this became the cache and that the struct below is immutable. I'll grab some coffee.

Reply:
It would be better to bring it to full immutability to guarantee safety, but this is the quickest way to working so it's worth it for now. The plan is to probably have a GSoC to follow up and do more stiffness estimates, in which case we'll pick this back up and probably make it immutable, but at least if it's fully thread-safe already that can wait.